app/assets/javascripts/views/export_csv_view.js

```diff
@@ -0,0 +1,48 @@
+// Handles getting info about bulk media thresholds
+ELMO.Views.ExportCsvView = class ExportCsvView extends ELMO.Views.ApplicationView {
+  get events() {
+    return {
+      'click #response_csv_export_options_download_media': 'calculateMediaSize'
+    };
+  }
+
+  initialize(params) {
+    $(".calculating-info").hide();
+    $(".error-info").hide();
+    $(".media-info").hide();
+  }
+
+  calculateMediaSize(event) {
+    if (($(event.target)).is(':checked')) {
+      $(".calculating-info").show();
+      $("input[type=submit]").prop("disabled", true);
+      this.spaceLeft();
+    } else {
+      $(".calculating-info").hide();
+      $(".error-info").hide();
+      $("input[type=submit]").removeAttr("disabled");
+      $(".media-info").hide();
+    }
+  }
+
+  spaceLeft() {
+    $.ajax({
+      url: ELMO.app.url_builder.build("media-size"),
+      method: "get",
+      data: "",
+      success: (data) => {
+        $(".calculating-info").hide();
```

Comment:
For readability and reliability, I like to delegate "ownership" of specific DOM elements to specific methods -- e.g. the `.calculating-info` could always be hidden/shown in the `spaceLeft` method so the logic is self-contained. That way you know that both the `calculateMediaSize` AND `spaceLeft` method are "pure" and don't leave DOM state half-way modified with the assumption that some other method will finish modifying it later (easy for bugs to creep in that way).

So my specific suggestion here would be to move `$(".calculating-info").show();` into this method above `ajax`, and do similar tweaks for other elements if reasonable. It doesn't need to be perfect.

Reply:
thanks for this Kevin! super helpful to learn.